### ISSUE TITLE
Why not use kwargs for options ?

### ISSUE THREAD

**Issue by althonos** [ManrajGrover/halo#10]
Hi there, this is a question / enhancement proposal:

instead of relying on type checking, why not use keyword arguments to initialize a `Halo` instance ? This would be more Pythonic IMO (although, I realize filing an issue just to tell you that might make me sound like an a**hole).

This way, you would have more control about accepted arguments (nothing stops someone from creating an `option` dictionnary with nonsensical options, or maybe just a typo), and also expose the default values of arguments to the documentation / inline help of `Halo`:

```python

class Halo(object):

    def __init__(self, text='', color='cyan', interval=-1, enabled=True, stream=sys.stdout):
        # ... #
```

Of course, this may cause problems for `interval` since it depends on the chosen spinner, but using `None` or **-1** would probably do the trick.

Once again, this is just a suggestion !

**Comment by dgovil** [comment]
I second this, it would make the API much easier and cleaner to use.

Some reasons:

- You only have a few arguments anyway so using a dictionary doesn't give you any benefits
- It's much easier to catch errors with actual parameters, because a typo won't break the code. Using dictionaries, a typo could be hidden for a long time.
- It's better in editors where auto complete and intellisense will tell you what parameters a callable takes. 
   That means you don't need to read the documentation just to remember each of the options.

**Comment by wookayin** [comment]
+1, Definitely. Please be pythonic.

**Comment by ManrajGrover** [comment]
@althonos I agree with you. @EVasseure was quick with the PR #13 too. Like I said before, I would love to incorporate this. 😄 Thank you for the suggestion. Keep them coming.

**Comment by ManrajGrover** [comment]
Fixed in 2d291bc
